src/sentry/static/sentry/app/routes.jsx

```diff
@@ -1151,40 +1153,119 @@ function routes() {
           <IndexRoute component={errorHandler(ProjectInstallOverview)} />
           <Route path=":platform/" component={errorHandler(ProjectInstallPlatform)} />
         </Route>
-
-        <Redirect from=":projectId/" to="/organizations/:orgId/issues/" />
-        <Redirect from=":projectId/issues/" to="/organizations/:orgId/issues/" />
-        <Redirect
-          from=":projectId/searches/:searchId/"
-          to="/organizations/:orgId/issues/"
-        />
-        <Redirect from=":projectId/dashboard/" to="/organizations/:orgId/dashboards/" />
-        <Redirect from=":projectId/releases/" to="/organizations/:orgId/releases/" />
-        <Redirect
-          from=":projectId/releases/:version/"
-          to="/organizations/:orgId/releases/:version/"
-        />
-        <Redirect
-          from=":projectId/releases/:version/new-events/"
-          to="/organizations/:orgId/releases/:version/new-events/"
-        />
-        <Redirect
-          from=":projectId/releases/:version/all-events/"
-          to="/organizations/:orgId/releases/:version/all-events/"
-        />
-        <Redirect
-          from=":projectId/releases/:version/artifacts/"
-          to="/organizations/:orgId/releases/:version/artifacts/"
-        />
-        <Redirect
-          from=":projectId/releases/:version/commits/"
-          to="/organizations/:orgId/releases/:version/commits/"
-        />
-
-        <Redirect
-          from=":projectId/user-feedback/"
-          to="/organizations/:orgId/user-feedback/"
-        />
+        <Route path=":projectSlug/">
+          {/* Support for legacy URLs (pre-Sentry 10). We just redirect users to new canonical URLs. */}
+          <IndexRoute
+            component={errorHandler(
+              redirectLegacyProjectRoute(
+                ({orgId, projectId}) =>
+                  `/organizations/${orgId}/issues/?project=${projectId}`
+              )
+            )}
+          />
+          <Route
+            path="issues/"
+            component={errorHandler(
+              redirectLegacyProjectRoute(
+                ({orgId, projectId}) =>
+                  `/organizations/${orgId}/issues/?project=${projectId}`
+              )
+            )}
+          />
+          <Route
+            path="searches/:searchId/"
+            component={errorHandler(
+              redirectLegacyProjectSavedSearchRoute(({orgId, projectId, searchQuery}) => {
+                return {
+                  pathname: `/organizations/${orgId}/issues/`,
+                  query: {...searchQuery, project: projectId},
+                };
+              })
+            )}
+          />
+          <Route
+            path="dashboard/"
+            component={errorHandler(
+              redirectLegacyProjectRoute(
+                ({orgId, projectId}) =>
+                  `/organizations/${orgId}/dashboards/?project=${projectId}`
```

Comment:
There's no 1:1 view for this (dashboards is a business only feature) - I think we'll want to just redirect them to the issues page.

Reply:
Oh interesting. I'll remove this.